frontend/src/bundles/admin-panel/admin-panel-api.ts

```diff
@@ -0,0 +1,116 @@
+import {
+    ApiPath,
+    ContentType,
+    UserDetailsApiPath,
+} from '~/bundles/common/enums/enums.js';
+import { HttpApiBase } from '~/framework/api/api.js';
+import { type Http } from '~/framework/http/http.js';
+import { type Storage } from '~/framework/storage/storage.js';
+
+import {
+    type UserDetailsFullResponseDto,
+    type UserDetailsShortResponseDto,
+} from './types/types.js';
+
+type Constructor = {
+    baseUrl: string;
+    http: Http;
+    storage: Storage;
+};
+
+class AdminApi extends HttpApiBase {
+    public constructor({ baseUrl, http, storage }: Constructor) {
+        super({ path: ApiPath.USER_DETAILS, baseUrl, http, storage });
+    }
+
+    // public async getUserDetailsByUserId(
+    //     payload: Partial<UserDetailsFullResponseDto>,
+    // ): Promise<UserDetailsFullResponseDto | null> {
+    //     const { userId = '' } = payload;
+
+    //     const response = await this.load(
+    //         this.getFullEndpoint('/', userId, {}),
+    //         {
+    //             method: 'GET',
+    //             contentType: ContentType.JSON,
+    //             hasAuth: true,
+    //         },
+    //     );
+    //     return response.json<UserDetailsFullResponseDto>();
+    // }
+
+    public async getShortUserDetailsByRole(payload: {
+        role: 'talent' | 'employer';
+    }): Promise<UserDetailsShortResponseDto[]> {
+        const { role = '' } = payload;
+
+        const response = await this.load(
+            this.getFullEndpoint(
+                UserDetailsApiPath.SHORT,
+                '?userType=',
+                role,
+                {},
```

Comment:
I think we should have query params for options which we provide as second params of this function

Reply:
Created helper for it